docs/reference/fields.txt

```diff
@@ -1079,3 +1081,70 @@ you can include the short versions of the modules.
      include Mongoid::Document
      include Mongoid::Timestamps::Updated::Short # For u_at only.
    end
+
+
+.. _field-names-with-periods-and-dollar-signs:
+
+Field Names with Periods (``.``) and Dollar Signs (``$``)
+=========================================================
+
+Mongo 5.0 `adds specialized support
+<https://docs.mongodb.com/manual/core/dot-dollar-considerations/#std-label-crud-concepts-dot-dollar-considerations>`_
+for field names that are prefixed with a dollar sign (``$``) or
+that contain periods (``.``).
+
+Fields prefixed with ``$`` or containing ``.`` are not directly accessible in standard queries.
+They may only be accessed with special database helper methods (new in MongoDB 5.0)
+including ``$getField``, ``$setField``, and ``$literal``.
+In normal usage scenarios, such fields behave as "hidden" fields on the document.
+In addition, they come with severe limitations, such as the inability
+to be indexed.
+
+Support in Mongoid
+------------------
+
+This section outlines the behavior of Mongoid and the Mongo Ruby Driver
+when working with fields whose names are prefixed with ``$`` or contain ``.``
+("specially-named fields"):
+
+* Mongoid's :ref:`field <field-types>` macro does not support defining specially-named fields.
```

Comment:
This is horrendous.

Reply:
Yes, but it doesn't support saving to the database today, does it?

Reply:
This is a driver limitation. Once it is lifted in the driver, I expect all published Mongoid versions will be able to persist fields like this.

Reply:
@p-mongo from your comments on this issue it appears you are fundamentally misunderstanding how these fields work in MongoDB, and furthermore you have a mistaken conception that they can be treated like regular fields in Mongoid to provide a "seamless" experience to the end-user.

Although such an approach may be **possible** to some extent, it will require so many hacks and caveats that will not be **practical**. The fields should be handled specially and only accessed via specialized `get`/set` functions.